extension/sigv4authextension/README.md

```diff
@@ -22,13 +22,23 @@ The configuration fields are as follows:
   * `session_name`: **Optional**. The name of a role session
   * `sts_region`: The AWS region where STS is used to assumed the configured role
     * Note that if a role is intended to be assumed, and `sts_region` is not provided, then `sts_region` will default to the value for `region` if `region` is provided
+* `region`: **Optional**. The AWS region for the service you are exporting to for AWS Sigv4. This is differentiated from `sts_region` to handle cross region authentication
+* `assume_role_with_web_identity`: **Optional**. Specifies the configuration needed to assume a role with web identity
+  * `arn`: The Amazon Resource Name (ARN) of a role to assume - this must have a trust policy to allow it to be assumed
+  * `token_file`: The path to the file containing the JWT token to be excahanged
```

Comment:
This seems to be the only difference from the `assume_role` configuration.  Would it make sense to have `token_file` as an optional field in the `assume_role` configuration and use its presence to determine how to assume the role?

Reply:
This is how the [AWS CLI is configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-configure-role-oidc).  The presence of the `web_identity_token_file` attribute in a profile indicates that the role should be assumed via that web identity.

Reply:
OK, I can implement that